### ISSUE TITLE
DEMO - Tyrant Not Recommended During Second Small Tyrant Window

### ISSUE THREAD

**Issue by Rynitaca** [Hekili/hekili#3371]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

The addon is not recommending Tyrant during the window of small dogs and Vile fiend (when Grimoire is on CD). I assume this is because it missed that Tyrant is now a 1 minute CD therefore a smaller tyrant window can be implemented in between the large Tyrant window that includes Grimoire.

### How to Reproduce

1. Enter the game as a DEMO Warlock
2. Perform the first opener in combat to set up the big Tyrant (vilefiend, dogs, Grimoire)
3. Wait until the next vilefiend is available (it also won't be recommended) 
4. Press vilefiend and dogs
5. Tyrant won't be recommended to extend them

### Snapshot (Link)

https://pastebin.com/8jRmYLK6

### Contact Information

sym1390

**Comment by Hekili** [comment]
The sim comparison would be helpful in this case.

**Comment by Hekili** [comment]
...or not.

![image](https://github.com/user-attachments/assets/72f0492b-64e4-46a5-a19c-1d87ebab3ac1)

It is likely that we'll need to wait for actual Warlock priorities to be published.

**Comment by Hekili** [comment]
This is convoluted, but...

```
        List ( tyrant ) called from ( Demonology:default:29 ) would FAIL at 0.00.
        ( cooldown.summon_demonic_tyrant.remains[0.00] < 15 & cooldown.summon_vilefiend.remains[22.71] < gcd.max[1.23] * 5 & cooldown.call_dreadstalkers.remains[7.75] < gcd.max[1.23] * 5 & ( cooldown.grimoire_felguard.remains[45.15] < 10 | ! set_bonus.tier30_2pc[0.00] ) & ( cooldown.summon_demonic_tyrant.remains_expected[0.00] < 15 | fight_remains[326.42] < 40 | buff.power_infusion.up[false] ) & ( ! raid_event.adds.exists[true] | raid_event.adds.exists[true] & raid_event.adds.remains[326.42] > 20 ) | 
        talent.summon_vilefiend.enabled[true] & cooldown.summon_demonic_tyrant.remains[0.00] < 15 & cooldown.summon_vilefiend.remains[22.71] < gcd.max[1.23] * 5 & cooldown.call_dreadstalkers.remains[7.75] < gcd.max[1.23] * 5 & ( cooldown.grimoire_felguard.remains[45.15] < 10 | ! set_bonus.tier30_2pc[0.00] ) & ( cooldown.summon_demonic_tyrant.remains_expected[0.00] < 15 | fight_remains[326.42] < 40 | buff.power_infusion.up[false] ) & ( ! raid_event.adds.exists[true] | raid_event.adds.exists[true] & raid_event.adds.remains[326.42] > 20 ) ) | 
        ( cooldown.summon_demonic_tyrant.remains[0.00] < 15 & ( buff.vilefiend.up[false] | ! talent.summon_vilefiend.enabled[true] & ( buff.grimoire_felguard.up[false] | cooldown.grimoire_felguard.up[false] | ! set_bonus.tier30_2pc[0.00] ) ) & ( cooldown.summon_demonic_tyrant.remains_expected[0.00] < 15 | buff.grimoire_felguard.up[false] | fight_remains[326.42] < 40 | buff.power_infusion.up[false] ) & ( ! raid_event.adds.exists[true] | raid_event.adds.exists[true] & raid_event.adds.remains[326.42] > 20 ) )
```

Your Vilefiend still had 22.71 seconds on its cooldown in your snapshot.  If you can get another snapshot where Vilefiend has less than 5 GCDs left, I can try to break down whatever else might be an issue.  I'm guessing there needs to be some threshold determined where you'd wait for Grimoire Felguard before Tyrant again.

`cooldown.grimoire_felguard.remains[45.15] < 10`

There will likely need to be an additional `cooldown.grimoire_felguard.remains > cooldown.summon_demonic_tyrant.remains + 60` or something along those lines.  I'll drop that in and see if it makes a difference.

**Comment by Hekili** [comment]
This was likely due to Charhound/Gloomhound not being detected in place of Vilefiend.  Should be resolved with 0.9.1b or later.
